java/src/main/java/com/google/appengine/tools/pipeline/impl/servlets/DeleteJobHandler.java

```diff
@@ -16,25 +16,27 @@
 
 import com.google.appengine.tools.pipeline.NoSuchObjectException;
 import com.google.appengine.tools.pipeline.PipelineRunner;
-import com.google.appengine.tools.pipeline.impl.PipelineManager;
-import lombok.RequiredArgsConstructor;
+import lombok.AllArgsConstructor;
 
 import java.io.IOException;
 
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 
+import javax.inject.Inject;
+
 /**
  * @author ozarov@google.com (Arie Ozarov)
  */
-@RequiredArgsConstructor
+@AllArgsConstructor
 public class DeleteJobHandler {
 
   public static final String PATH_COMPONENT = "rpc/delete";
   private static final String ROOT_PIPELINE_ID = "root_pipeline_id";
 
-  private final PipelineRunner pipelineManager;
+  @Inject
+  PipelineRunner pipelineManager;
```

Comment:
Mmm, ok but what is happening? I'm assuming that with `RequiredArgsConstructor` + *final fields* was failing, not sure if it was failing because the class cannot be created or if the arguments passed to the constructor were null.

Now is with `AllArgsConstructor` + *field*; I'd say that it is still using the constructor for building the object with dependency and not property injection, right?

Then, couple of questions: 
1) what happen if you keep `@RequiredArgsConstructor` + *final fields* with marked them with *@Inject* ?
2) what happen if you declare a constructor marked with `@Inject` with all *final fields* declared? The cons of this approach is that we need to write it ourselves, as we cannot use lombok for that.

The only thing I dislike with current approach (in case we are using constructor, or even property) is that we are exposing the field as a package visibility. I think is cleaner to do it with private final, but if is done in that way, it will not work with dagger when using property injection. 

So in such case, 3) can we put ... 
```suggestion
private final PipelineRunner pipelineManager;
```

or 
```suggestion
private PipelineRunner pipelineManager;
```
?

Reply:
so ... it's complicated. a couple different issues that we should think about it.

- Dagger can't inject `private` fields
- these are serlvets, so implement `Serializable` and in theory, if not in practice, could be serialized/deserialized between requests. So this means either dependencies must be serialized too (complicated) or must be `transient` and we use field-injection


that said, long-term we won't use servlets anyways ...

Reply:
so issues; will try to just make it work ...